rioxarray/raster_writer.py

```diff
@@ -155,7 +155,9 @@ def to_raster(self, xarray_dataarray, tags, windowed, lock, compute, **kwargs):
         **kwargs
             Keyword arguments to pass into writing the raster.
         """
-        dtype = kwargs["dtype"]
+        numpy_dtype = kwargs["dtype"]
+        if xarray_dataarray.encoding.get("_rasterio_dtype") == "complex_int16":
```

Comment:
I am thinking we should check if the numpy dtype is complex in addition to the rasterio dtype check here. Just in case they did an `astype(...)` or passed in a different dtype to convert it to something else.